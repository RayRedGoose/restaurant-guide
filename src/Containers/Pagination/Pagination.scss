@import "../../assets/css/variables.scss";
@import "../../assets/css/index.scss";

.pagination {
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 3rem 0;
  .pages {
    margin: 0 2rem;
    color: $accent-color;
    font-family: $accent-font;
    font-size: 1.5rem;
    font-weight: 700;
    letter-spacing: 0.05em;
  }
}
