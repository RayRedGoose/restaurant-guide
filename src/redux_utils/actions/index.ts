import {
  IRestaurantObject,
  IAddRestaurantAction
} from 'assets/ts/interfaces';

export const addRestaurants = (restaurants: IRestaurantObject[]): IAddRestaurantAction => ({
  type: 'ADD_RESTAURANTS',
  restaurants: restaurants
});
