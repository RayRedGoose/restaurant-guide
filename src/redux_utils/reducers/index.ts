import { combineReducers } from 'redux';
import restaurants from './restaurants/restaurants';

const rootReducer = combineReducers({
  restaurants
});

export default rootReducer;
