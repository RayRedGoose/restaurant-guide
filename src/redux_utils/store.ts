import { Store, createStore } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";
import rootReducer from "./reducers";
import { IAppStore } from "assets/ts/interfaces";

const store: Store<IAppStore> = createStore(rootReducer, composeWithDevTools());

export default store;
