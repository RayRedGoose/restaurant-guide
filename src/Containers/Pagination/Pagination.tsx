import "./Pagination.scss";
import React from "react";
import { useDispatch, useSelector } from "react-redux";
import { IAppStore } from "assets/ts/interfaces";
import { changeCurrentPage } from "redux_utils/actions";

const Pagination: React.FC = () => {
  const dispatch = useDispatch();
  const { displayPage, maxPages } = useSelector((store: IAppStore) => ({
    displayPage: store.currentPage + 1,
    maxPages: store.maxPages,
  }));

  const decreaseCounter = (): void => {
    if (displayPage > 1) dispatch(changeCurrentPage(-1));
  };

  const increaseCounter = (): void => {
    if (displayPage < maxPages) dispatch(changeCurrentPage(1));
  };

  return (
    <footer className="pagination">
      <button
        className="pagination-btn"
        onClick={decreaseCounter}
        style={{ opacity: `${displayPage > 1 ? 1 : 0}` }}
      >
        Previous page
      </button>
      <p className="pages">{`${displayPage} / ${maxPages}`}</p>
      <button
        className="pagination-btn"
        onClick={increaseCounter}
        style={{ opacity: `${displayPage < maxPages ? 1 : 0}` }}
      >
        Next page
      </button>
    </footer>
  );
};

export default Pagination;
