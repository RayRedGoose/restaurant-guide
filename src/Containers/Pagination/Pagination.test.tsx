import React from 'react';
import { shallow } from 'enzyme';
import Pagination from './Pagination';
import { changeCurrentPage } from 'redux_utils/actions';

const mockDispatch = jest.fn();
let mockPages = {
  displayPage: 2,
  maxPages: 6
};

jest.mock('react-redux', () => ({
  useDispatch: () => mockDispatch,
  useSelector: () => mockPages
}));
jest.mock('redux_utils/actions');

describe("Pagination", () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = shallow(<Pagination />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should use data from the store', () => {
    const text = wrapper.find('.pages');
    expect(text.text()).toEqual('2 / 6');
  });

  describe("Click Events", () => {
    it('should call changeCurrentPage with -1 when previous page button is clicked', () => {
      wrapper.find('#prev-page-btn').simulate('click');
      expect(changeCurrentPage).toHaveBeenCalledWith(-1);
    });

    it('should call changeCurrentPage with 1 when next page button is clicked', () => {
      wrapper.find('#next-page-btn').simulate('click');
      expect(changeCurrentPage).toHaveBeenCalledWith(1);
    });

    it('should hide previous page button when number of page equal 1', () => {
      mockPages = {
        displayPage: 1,
        maxPages: 6
      };
      const wrapper = shallow(<Pagination />);
      const button = wrapper.find('#prev-page-btn');
      expect(button).toHaveLength(0);
    });

    it('should hide next page button when number of page equal maxPages', () => {
      mockPages = {
        displayPage: 6,
        maxPages: 6
      };
      const wrapper = shallow(<Pagination />);
      const button = wrapper.find('#next-page-btn');
      expect(button).toHaveLength(0);
    });
  });
});
