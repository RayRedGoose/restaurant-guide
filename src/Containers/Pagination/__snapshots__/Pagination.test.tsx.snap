// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should match the snapshot 1`] = `
<footer>
  <button
    id="prev-page-btn"
    onClick={[Function]}
  >
     - 
  </button>
  <p
    className="pages"
  >
    2 / 6
  </p>
  <button
    id="next-page-btn"
    onClick={[Function]}
  >
     + 
  </button>
</footer>
`;
